openedx/core/lib/request_utils.py

```diff
@@ -95,17 +95,27 @@ class CookieMonitoringMiddleware:
     Middleware for monitoring the size and growth of all our cookies, to see if
     we're running into browser limits.
     """
+    COOKIE_MONITORING_REQUEST_CACHE_KEY = "CookieMonitoringMiddleware"
+    COOKIE_HEADER_LOG_MESSAGE_KEY = "cookie_header_log_message"
 
     def __init__(self, get_response):
         self.get_response = get_response
 
     def __call__(self, request):
-        response = self.get_response(request)
-        # monitor after response to ensure logging can include user id where possible.
+        # Monitor at request-time to skip any cookies that may be added during the request.
         try:
             self.log_and_monitor_cookies(request)
```

Comment:
How would you feel about instead returning a string to log so that we can avoid the request cache code?

Reply:
@timmc-edx: I fixed this. I was so used to the pattern of using RequestCache for old middleware where `process_request` and `process_response` were separate methods, that it didn't occur to me that this was a single method.